datafusion/datasource-parquet/src/opener.rs

```diff
@@ -384,6 +353,24 @@ impl FileOpener for ParquetOpener {
                 .map(move |maybe_batch| {
                     maybe_batch
                         .and_then(|b| schema_mapping.map_batch(b).map_err(Into::into))
+                })
+                .take_while(move |_| {
+                    if let Some(file_pruner) = file_pruner.as_ref() {
+                        match file_pruner.should_prune() {
```

Comment:
This is basically applying the filter on each record batch, right? 

I think once we can actually push the filters into the parquet scan (which I realize I have been talking about for months...) this could become be entirely redundant

On the other hand, it also stops the input immediately if we find out the file should stop 🤔

Reply:
Right the point is the _stopping_ which the parquet pruning will _not_ do!